### PR TITLE
An example notebook of deep learning workflow

### PR DESCRIPTION
**Pull request by geek-yang**
We need an example deep learning workflow to show how `s2spy` and `lilio` can facilitate the workflow, which includes:
- LSTM network
- Transformer with multi-head attention (working progress)

In next PR, we can put deep learning examples and a simple linear regression model to show that with `s2spy` and `lilio` one can easily obtain a baseline with regression forecasting (see issue #5).

Note that the transformer notebook is still a working progress and it needs to be fine-tuned to get sensible results (see issue #6). This will be achieved in another PR.

To review this PR, just check the notebook `workflow/pred_temperature_LSTM.ipynb`.

### CODE REVIEW

File: src/transformer.py
Comment (semvijverberg) [inline src/transformer.py:1]: I would love to go through this script in-person to understand it in more detail, for I can only state it looks super fancy.

File: setup.cfg
Comment (semvijverberg) [inline setup.cfg:1]: This is nicely done

File: workflow/pred_temperature_ridge.ipynb
Comment (semvijverberg) [inline workflow/pred_temperature_ridge.ipynb:1]: As mentioned earlier, would kind of be nice to copy the calendar and CV method across the 3 notebook so that the methods can be compared.

File: workflow/pred_temperature_transformer.ipynb
Comment (semvijverberg) [inline workflow/pred_temperature_transformer.ipynb:1]: Same comment as above.

And I guess this is not working yet? I saw that the validation/train loss remained constant over the 10 epochs (<- low number)

**Review by semvijverberg — CHANGES_REQUESTED** [review]

File: workflow/pred_temperature_ridge.ipynb
Comment (geek-yang) [reply workflow/pred_temperature_ridge.ipynb:1]: Yes, right. Therefore I created an issue for it #5 and I will work on it in another PR as it needs some more effort (e.g. some changes to the metrics for comparison), otherwise this PR gets too big.

File: workflow/pred_temperature_transformer.ipynb
Comment (geek-yang) [reply workflow/pred_temperature_transformer.ipynb:1]: This is still working progress. As the PR gets too big, I prefer to fix it in another one. An issue is created for it #6.

**Comment by geek-yang** [comment]
> Ah oeps, only review `workflow/pred_temperature_LSTM.ipynb`. I should have read the text!

No worry. If `workflow/pred_temperature_LSTM.ipynb` looks nice to you, could you please give me a go? For the rest, I have created some issues and I plan to open another PR for them since this one is getting too big. Thanks.

**Comment by semvijverberg** [comment]
Oke Yang,

Yes the LSTM looks very nice! As commented, I prefer to use the same calendar throughout and I prefer the LSTM calendar :). Maybe copy this calendar to the Ridge one.

**Comment by geek-yang** [comment]
> Oke Yang,
> 
> Yes the LSTM looks very nice! As commented, I prefer to use the same calendar throughout and I prefer the LSTM calendar :). Maybe copy this calendar to the Ridge one.

Sounds good. I copy the same calendar from LSTM to Ridge. Actually my plan is to keep these notebooks, and later add another one to put them together for the comparison purpose. Since the comparison notebook would be quite long, it will be nice for the user to start with a simpler one, which contains only our regression example or the deep learning example, and then move on to the comparison one.

Besides, we also need to define the same task and metrics for the comparison case, which needs some thinking.

**Comment by semvijverberg** [comment]
> > Oke Yang,
> > Yes the LSTM looks very nice! As commented, I prefer to use the same calendar throughout and I prefer the LSTM calendar :). Maybe copy this calendar to the Ridge one.
> 
> Sounds good. I copy the same calendar from LSTM to Ridge. Actually my plan is to keep these notebooks, and later add another one to put them together for the comparison purpose. Since the comparison notebook would be quite long, it will be nice for the user to start with a simpler one, which contains only our regression example or the deep learning example, and then move on to the comparison one.
> 
> Besides, we also need to define the same task and metrics for the comparison case, which needs some thinking.

That sounds good!
